src/record.jl

```diff
@@ -10,8 +10,8 @@ function Recording(f1, f2, sim::NestedFilterSimulation)
     return Recording(f1, f2, data)
 end
 
-function update!(rec::Recording, sim)
-    push!(rec.data, rec.f1(sim))
+function update!(rec::Recording, sim, i, time)
```

Comment:
What is `i` needed for? I can see that `time` is needed because that is not stored in `sim`.

Reply:
It is meant to modify the vector of stored values, something like 
my_values = zertos(T)
rec.my_values[i] = sim.value[i]
Shall we use something more efficient, like a "append" function ?

Reply:
update! uses push!, so this is not needed